katdal/concatdata.py

```diff
@@ -233,11 +232,13 @@ def __init__(self, data):
 
     def __getitem__(self, key):
         """Extract timestamp, value and status of each sensor data point."""
-        return np.concatenate([sd[key] for sd in self.data])
+        return np.concatenate([sd[key] for sd in self._data])
 
-    def __len__(self):
-        """Number of sensor data points."""
-        return np.sum([len(sd) for sd in self.data])
+    def __bool__(self):
+        """True if sensor has at least one data point."""
+        return any([bool(sd) for sd in self._data])
```

Comment:
Also, is this going to work given that `sd` is presumably an ndarray, on which `bool` doesn't do what one would normally assume?